lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1180,12 +1210,22 @@ private function isLimitQuery()
      * Converts this instance into an INSERT string in SQL.
      *
      * @return string
+     * @throws QueryException
      */
     private function getSQLForInsert()
     {
-        return 'INSERT INTO ' . $this->sqlParts['from']['table'] .
-        ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')' .
-        ' VALUES(' . implode(', ', $this->sqlParts['values']) . ')';
+        $statement = 'INSERT INTO ' . $this->sqlParts['into']['table'];
+        if( !empty($this->sqlParts['values']) && count( array_keys($this->sqlParts['values']) ) > 0 ) {
+            $statement .= ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')';
+        }
+
+        if( !empty($this->sqlParts['select'] ) ) {
+            $statement .= ' (' . $this->getSQLForSelect() . ')';
+        }
+        else {
```

Comment:
What if both `'select'` and `'values'` parts are defined?

Reply:
then a 'normal' inster statement is generated. You might even throw an exception if both are specified

Reply:
Throwing is a better idea here